### ISSUE TITLE
[2.4.3] UI - Unable to access upgrade page if used chart version doesn't exist anymore.

### ISSUE THREAD

**Issue by Lebvanih** [rancher/rancher#27025]
We are not too sure if this is a bad usage of versioning on our side or a real bug, but the behavior changed when upgrading from 2.3.x to 2.4.3.

**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**

1. Create a dummy chart with `helm create testapp`
2. Edit the apiVersion and switch it to v1. Keep the default version provided
3. Import the chart in Rancher (we use a gitlab instance to host them) and deploy the apps. Then click on Upgrade and the page properly load allowing to change version/parameters.
4. Update the version field in the chart, repush to gitlab and refresh the apps in rancher (in Apps > Launch > Refresh).

**Result:**
If you try to upgrade after that last refresh of apps, even if rancher see the new version, the page will not load, it will show the loading animation after clicking on upgrade then return the the app dashboard without allowing you yo upgrade. This was working previously, even if we would encounter sometimes a message saying that the version doesn't exist anymore, it would default to the last version at least.

**Other details that may be helpful:**
We tested multiple pattern, and it seems that it come from the fact that the version in use is overwritten. If we do versioning via a directory it will work for the upgrade unless we delete the directory of the version currently in use. Example:
0.1/Charts.yml refer to version 0.1
0.1-beta/Charts.yam refer to version 0.1-beta (in Use)
0.2/Charts.yml refer to version 0.2

The upgrade from 0.1 to 0.1-beta would work at the start. After we add the 0.2 version it still work, howeverIf we delete the directory of the 0.1-beta to clean up a bit, this trigger the same issue (as in, Rancher detect the upgrade, but is not able to open the upgrade page at all). There is no logs visible in the console of the browser, and I can't find anything specials in the logs.

We are unsure if this change of behavior is to enforce proper usage of versioning, however, we would expect at least an error or an easy way to get out of that issue. Currently the only way we found to find it and get access back to the upgrade page, is to edit via api the externalId to a version that still exist..

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.4.3
- Installation option (single install/HA): HA

**Comment by mschiessl** [comment]
I just had a similar issue.
After reading the chart.yaml documentation i found that the chart version requires a full SemVer style version (documented here: https://semver.org/)
After i changed my versions from 1.0 and 1.1 towards 1.0.0 and 1.1.0 (path and cahrt !) it worked again for me.
Dunno if htis was just luck or really is the root cause - but anyhow maybe you want to give it a shot

**Comment by Lebvanih** [comment]
I am pretty sure I tested a similar flow on my first tests, because we suspected our version number not being SemVer compliant. The version of our application on which we had the issue was 2020.2-184. 

When we renamed it to 2020.2.184 it started to work again. But when I did my test before creating the ticket, we noticed that 2020.2-184 if deployed as first, would still be working unless we overwrite the initial chart with a higher version (we were not using versioned directory which is a bad practice we know). And that's how I came up after with the replicate step here. 

It seems that if a chart is version is used, and if you overwrite it with a new version or remove it (via cleaning). The application linked to it can't be upgraded anymore and you are forced to edit via API the version number to unlock it.
